mmdet/engine/runner/loops.py

```diff
@@ -37,3 +40,104 @@ def run(self):
+
+        assert len(outputs) == 1, 'During inference, only support 1 batch '
+        'size per gpu for now in video tasks.'
+
+        track_data_sample = outputs[0]
+        ori_video_len = track_data_sample.ori_video_length
+        if len(track_data_sample) == ori_video_len:
+            self.evaluator.process(data_samples=outputs, data_batch=data_batch)
+        else:
+            frame_id = track_data_sample[0].frame_id
+            if frame_id == 0:
+                self.video_outputs = track_data_sample.clone()
+            else:
+                self.video_outputs.video_data_samples.append(
+                    track_data_sample[0].clone())
+                self.video_outputs.set_metainfo(
+                    dict(video_length=self.video_outputs.video_length + 1))
+                if frame_id == ori_video_len - 1:
+                    self.evaluator.process(
+                        data_samples=[self.video_outputs],
+                        data_batch=data_batch)
+
+        self.runner.call_hook(
+            'after_val_iter',
+            batch_idx=idx,
+            data_batch=data_batch,
+            outputs=outputs)
+
+
+@LOOPS.register_module()
+class VideoTestLoop(TestLoop):
+    """Loop for test of video model.
+
+    The difference between this loop and ``TestLoop`` is that this loop does
+    not pass the results from ``predict`` in model into the evaluator util the
+    whole video is processed.
+    """
+
+    @torch.no_grad()
+    def run_iter(self, idx, data_batch: Sequence[dict]) -> None:
+        """Iterate one mini-batch.
+
+        Args:
+            data_batch (Sequence[dict]): Batch of data from dataloader.
+        """
+        self.runner.call_hook(
+            'before_test_iter', batch_idx=idx, data_batch=data_batch)
+        # predictions should be sequence of BaseDataElement
+        with autocast(enabled=self.fp16):
+            outputs = self.runner.model.test_step(data_batch)
+
+        assert len(outputs) == 1, 'During inference, only support 1 batch '
+        'size per gpu for now in video tasks.'
+
+        track_data_sample = outputs[0]
+        ori_video_len = track_data_sample.ori_video_length
+        if len(track_data_sample) == ori_video_len:
+            self.evaluator.process(data_samples=outputs, data_batch=data_batch)
+        else:
+            frame_id = track_data_sample[0].frame_id
+            if frame_id == 0:
+                self.video_outputs = track_data_sample.clone()
+            else:
+                self.video_outputs.video_data_samples.append(
+                    track_data_sample[0].clone())
+                self.video_outputs.set_metainfo(
+                    dict(video_length=self.video_outputs.video_length + 1))
+                if frame_id == ori_video_len - 1:
+                    self.evaluator.process(
+                        data_samples=[self.video_outputs],
+                        data_batch=data_batch)
```

Comment:
Can you move this part to the evaluator or metric?

Reply:
Sounds better. We will support both video predictions and image predicts in the Metric and deprecates these video loops.

Reply:
You can leave a TODO here to avoid forgetting.

Reply:
Have removed that part of code.